src/fish.cpp

```diff
@@ -214,10 +222,23 @@ static int read_init(const struct config_paths_t &paths) {
     return 1;
 }
 
+int run_command_list(std::vector<std::string> *cmds, const io_chain_t &io) {
+    int res = 1;
+    parser_t &parser = parser_t::principal_parser();
+
+    for (size_t i = 0; i < cmds->size(); i++) {
+        const wcstring cmd_wcs = str2wcstring(cmds->at(i));
+        res = parser.eval(cmd_wcs, io, TOP);
+    }
+
+    return res;
+}
+
 /// Parse the argument list, return the index of the first non-flag arguments.
-static int fish_parse_opt(int argc, char **argv, std::vector<std::string> *cmds) {
-    static const char *short_opts = "+hilnvc:p:d:D:";
+static int fish_parse_opt(int argc, char **argv, command_line_switches_t *switches) {
```

Comment:
The class and var name don't accurately convey their purpose. How about `commands_to_run_t` and `cmds`?  Alternatively, to reflect the intent to fold the other flags into this structure and be consistent with the pattern used for internal commands it could be `fish_cmd_opts_t` and `opts`.